File: POTW/week-5/CS_Solution5.java
/*
 * Avinash Anand
 * Given a positive integer n (1 ≤ n ≤ 99999999), print out the largest prime factor of n.
 */

package potw;

import java.util.Scanner;

public class CS_Solution5 {

    public static int findMaximumPrimeFactor(int num) {
        for (int i = 2; i <= Math.floor(Math.sqrt(num)); i++) {
            while (num % i == 0) {
                if (num == i) {
                    break;
                }
                num /= i;
            }
        }
        return num;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int num = scanner.nextInt();
        System.out.println(findMaximumPrimeFactor(num));
    }

}

File: POTW/week-5/CS-Solution5.py
# Avinash Anand
# Given a positive integer n (1 ≤ n ≤ 99999999), print out the largest prime factor of n.

import math

def maximum(num):
    for i in range(2, math.floor(math.sqrt(num))+1):
        while num % i == 0:
            if num == i:
                break
            num /= i
    return int(num)

a = int(input())

print(maximum(a))